source/administration/performance-alerting-guide.rst

```diff
@@ -0,0 +1,121 @@
+
+1. In Mattermost:
+  a. Create an Alerts channel
+  b. Create an `incoming webhook <https://docs.mattermost.com/developer/webhooks-incoming.html>`_ for the Alerts channel and copy the URL
+
+2. In Grafana:
+  a. Under the alert icon in the sidebar, click “Notification channels”
+  b. Click “Add channel”
+  c. Enter “Mattermost Alerts Channel” as the name
+  d. For type, select “Slack”
+  e. Paste your webhook URL into the URL field
+  f. Include an @ mention in the mention field, if you want to send mentions when an alert is posted to Mattermost
+  g. Press “Send Test” to test the alert
+
+If you would also like to get email alerts, you can follow `these instructions <http://docs.grafana.org/alerting/notifications/#email.>`_ to set that up.
+
+**Configuring Alerts**
+
+The `Mattermost dashboards <https://grafana.com/dashboards?search=mattermost>`_ for Grafana come with some partially pre-configured alerts on the following charts:
+
+* CPU Utilization Rate
+* Memory Usage
+* Number of Goroutines
+* Number of API Errors per Second
+* Mean API Request Time
+
+To configure alerts 1) set an appropriate threshold and 2) enable notifications. Enabling notifications are the same for each chart, but setting the correct threshold can have some variances that are better handled on a per-chart basis.
+
+For each chart, click on the chart name and click “Edit”:
+
+.. image:: ../images/perf-1.png
+
+Then click on the “Alert” tab:
+
+.. image:: ../images/perf-2.png
+
+The alert threshold, which will be discussed in the sections below, is the last field under “Conditions” (the one set to 600 in the screenshot above).
+
+See the sections below for how to set the threshold for each individual chart. If you would like to add your own custom alert conditions, configure them here.
+
+.. image:: ../images/perf-3.png
+
+To enable the notifications for any alerts, click on the “Notification” tab on the left and select “Mattermost Alerts Channel” under “Send to”:
+
+.. image:: ../images/perf-4.png
+
+Fill in a message if you would like to add more context to the alert.
+
+By default, the alerts are configured to check the average of a chart over the last minute to see if that value is above a threshold. If it’s above the threshold, the alert will be triggered. Since it’s an average over the last minute, small spikes that go past the threshold won’t necessarily cause an alert. This helps prevent false positives that would result from natural spikes in usage. The alert state for each chart is evaluated every minute.
+
+The sections below describe each chart in more detail.
+
+CPU Utilization Rate
+---------------------
+
+CPU Utilization Rate is fairly straightforward. It tracks the CPU usage of the app servers as a percentage. The maximum percentage is based on the number of CPU cores or vCPUs your app server has. For example, if you have four CPU cores and your app server was at 100% utilization rate on all four cores, the graph would show 400% for that app server.
```

Comment:
`CPU Utilization Rate is fairly straightforward. It tracks...` --> `CPU Utilization Rate tracks...`